File: Lab2/Lab2.R
library(ggplot2)
library(dplyr)

setwd("C:/Users/Vasulevus/Documents/Github/R_Python_Labs/Lab2")
death <- read.csv('filmdeathcounts.csv')
#print(head(death))
#str(death)
death$body_per_min <- death$Body_Count / death$Length_Minutes

#print(head(death))

#print(
#    ggplot(death, aes(x=body_per_min)) + 
#    geom_histogram(bins = 20, color = "grey", fill = "lightblue")
#)

#print(
#death %>%
#    top_n(n = 10,Body_Count) %>%
#    arrange(desc(Body_Count))
#)
#
#print(
#    death %>%
#    top_n(n = 10, body_per_min) %>%
#    arrange(desc(body_per_min))
#)

#print(
#    ggplot(death, aes(x=IMDB_Rating)) + 
#    geom_histogram(bins = 10, color = "darkgrey", fill="lightgreen")
#)

#print(
#    death %>%
#    summarise(mean(IMDB_Rating), sd(IMDB_Rating))
#)
imdb_mean <- mean(death$IMDB_Rating)
imdb_sd <- sd(death$IMDB_Rating)
set.seed(900)
imdb_sim <- rnorm(n=nrow(death), mean = imdb_mean, sd = imdb_sd)

death$imdb_simulation <- imdb_sim

#print(
#    ggplot(death, aes(x=imdb_sim)) +
#    geom_histogram(bin=10, fill = "lightgreen", color = "black")
#)

print(
    ggplot(death, aes(sample = imdb_sim)) + stat_qq() #перевірка нормальності розподілу
)

print(ggplot(death,aes(sample=IMDB_Rating))+ stat_qq())

File: Lab2/Lab2.py
import pandas
import plotnine
import random
import numpy

movie_body_count = pandas.read_csv('filmdeathcounts.csv')
#print(movie_body_count)
movie_body_count.head() #перші п'ять рядків

movie_body_count.info() #структура



movie_body_count["Body_per_min"] = movie_body_count["Body_Count"] / movie_body_count["Length_Minutes"]

plotnine.ggplot(movie_body_count) + plotnine.aes(x = "Body_Count") + plotnine.geom_histogram(bins=20, color = "grey", fill = "lightblue")


movie_body_count.sort_values("Body_Count",ascending=False).head(10) #найбільша кількість тіл

movie_body_count.sort_values("Body_per_min", ascending=False).head(10)


plotnine.ggplot(movie_body_count) + plotnine.aes(x = "IMDB_Rating") + plotnine.geom_histogram(bins=10, color = "grey", fill = "lightblue")

imdb_mean = movie_body_count["IMDB_Rating"].mean()
imdb_sd = movie_body_count["IMDB_Rating"].std()

random.seed(900)

imdb_sim = numpy.random.normal(loc = imdb_mean, scale=imdb_sd, size= len(movie_body_count.index) ) #розрахунок нормального розподілу

movie_body_count["imdb_simulation"] = imdb_sim

plotnine.ggplot(movie_body_count) + plotnine.aes(x="imdb_simulation") + plotnine.geom_histogram(bins=10, color = "grey", fill = "lightblue") #візуалізація статистичного розподілу

plotnine.ggplot(movie_body_count) + plotnine.aes(sample = imdb_sim) + plotnine.stat_qq() #візуалізація нормального розподілу для статистичного нормального розподілу

plotnine.ggplot(movie_body_count) + plotnine.aes(sample = "IMDB_Rating") + plotnine.stat_qq() #візуалізація нормального розподілу для IMDB рейтингу